rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1227,18 +1259,32 @@ public BooleanWrapper nodeIsAvailable(String nodeUrl) {
 
             if (node == null) {
                 logger.warn("Cannot set node as available, the node is unknown: " + nodeUrl);
+                if (logger.isDebugEnabled()) {
+                    logger.warn("Known nodes are: " + Arrays.toString(allNodes.keySet().toArray()));
+                }
                 nodeUrlsNotKnownByTheRM.add(nodeUrl);
             } else if (node.isDown()) {
                 restoreNodeState(nodeUrl, node);
             } else {
                 if (logger.isDebugEnabled()) {
-                    logger.debug("The node identified by " + nodeUrl + " is known but not DOWN, no action performed");
+                    logger.debug("The node identified by " + nodeUrl + " is known and not DOWN, no action performed");
                 }
             }
         }
         return nodeUrlsNotKnownByTheRM.build();
     }
 
+    private void waitForRMCoreToBeInitialized() {
+        while (!rmCoreInitialized) {
+            try {
+                logger.info("Waiting for Resource Manager to be initialized");
+                countDownLatch.await();
+            } catch (InterruptedException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either re-interrupt this method or rethrow the "InterruptedException". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2142)